### ISSUE TITLE
Bambu Studio - Sprial Vase Mode - Producing "Seam/Artifact" on Lofted Surfaces at "Layer Change"

### ISSUE THREAD

**Issue by brandonraineri** [bambulab/BambuStudio#2744]
### Bambu Studio Version

1.7.7.89

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 10 Enterprise

### Printer

Bambu Lab X1

### How to reproduce

1.  Import a Part with a lofted surface that will be printed in vase mode
2.  Slice the Part
3.  Find the "Layer Change" location on the lofted part in the preview and notices it creates a "seam" or position change.
5. This "Seam/Defect" shows up on the printed part

### Actual results

I've tried this on multiple different models with lofted parts.  Tried importing extreme quality STL, STEP, etc. and still produces this "seam/artifact".  I've also tried different extrusion widths, layer heights, etc.  It only shows up on the lofted surface and does not show up if the surface is simply vertical/extruded.

![image](https://github.com/bambulab/BambuStudio/assets/109975882/eb9058e7-834c-404e-a8fe-44b7989f1748)

![IMG_3602](https://github.com/bambulab/BambuStudio/assets/109975882/cee36932-2c20-4554-bf99-10758183d2db)

![image](https://github.com/bambulab/BambuStudio/assets/109975882/c406ace3-6bac-4e6c-b450-deb873726850)

![image](https://github.com/bambulab/BambuStudio/assets/109975882/cf268b07-7a32-4577-b146-e3e53ba17f3f)

![image](https://github.com/bambulab/BambuStudio/assets/109975882/a06e3009-f45c-4765-a1ed-eac72555de46)

### Expected results

All walls should be smooth with no visible change in position

### Project file & Debug log uploads

[EGO Stubby Nozzle 70mm v1_ASA.zip](https://github.com/bambulab/BambuStudio/files/12915939/EGO.Stubby.Nozzle.70mm.v1_ASA.zip)

### Checklist of files to include

- [ ] Log file
- [X] Project file

**Comment by QingZhangBambu** [comment]
There is a sharp change between this layer's end and the next layer's start., because of the z change

**Comment by brandonraineri** [comment]
After doing more digging, it seems like this is a known bug in PS for some time now.  Seems like there are some scripts to post process the g-code but still somewhat experimental.

[https://github.com/prusa3d/PrusaSlicer/issues/2841](url)

**Comment by GuzziRaz** [comment]
> There is a sharp change between this layer's end and the next layer's start., because of the z change

That sharp change **is** the bug. A sudden Z-hop defeats the whole idea with spiral/vase mode. There should be a tiny increase of Z all around the object, thus no seam.

Correct link for Prusa bug: https://github.com/prusa3d/PrusaSlicer/issues/2841

**Comment by andrewboktor** [comment]
> > There is a sharp change between this layer's end and the next layer's start., because of the z change
> 
> That sharp change **is** the bug. A sudden Z-hop defeats the whole idea with spiral/vase mode. There should be a tiny increase of Z all around the object, thus no seam.
> 
> Correct link for Prusa bug: [prusa3d/PrusaSlicer#2841](https://github.com/prusa3d/PrusaSlicer/issues/2841)

This is not caused by a sudden change in Z, it's caused by a sudden change in XY since the layers don't stack perfectly on top of each other. So the nozzle moves horizontally from the end of one layer to the beginning of the next.

**Comment by QuiNz0r** [comment]
Please I urge you to fix this, I mainly print vases and this bug is driving me insane

**Comment by QuiNz0r** [comment]
> There is a sharp change between this layer's end and the next layer's start., because of the z change

That is the actual bug, there should be NO sharp Z change in Spiral vase mode!

**Comment by GuzziRaz** [comment]
> This is not caused by a sudden change in Z, it's caused by a sudden change in XY since the layers don't stack perfectly on top of each other. So the nozzle moves horizontally from the end of one layer to the beginning of the next.

This is correct. Still a bug though - or rather, the current spiral mode is a hack. I understand it's not a trivial thing.

Very interesting reasoning, ideas and workaround (also hacks, but promising) in https://github.com/prusa3d/PrusaSlicer/issues/2841

Quoting myself [from there](https://github.com/prusa3d/PrusaSlicer/issues/2841#issuecomment-1832184456):
> > Cura fixed it for most parts ages ago. There is an option called smoot spiralized contour. From what I see it just interpolates the XY coordinates of each point of one spiral according to the one above. The first point isn't moved at all, the last point is 100% moved to where the first of the next spiral will be. If it's a perfect cylinder no point will move.
> 
> Assuming we slice from bottom up, can we not just do the opposite? Then we only need to know the layer below (which we do because it's already sliced): Interpolate the XY coordinates of each point of one spiral according to the one below. The **first** point is 100% moved to where the last of the previous spiral was, the **last** point isn't moved at all. If it's a perfect cylinder no point will move.

**Comment by GuzziRaz** [comment]
Fixed in Orca - please merge!

**Comment by QingZhangBambu** [comment]
it will be added in 1.9 version

**Comment by m50434** [comment]
> it will be added in 1.9 version

At what time is the release of 1.9 ?
I really need this to print objects in spiral mode without z-seem!

Is there a 1.9 beta-download available?

**Comment by Tomsko567** [comment]
Hi, I resolved this by accident. I increased the count of bottom layers and the artifacting I had went away after re-slicing. This has worked for me both times I have had this problem. The second time I used 6 bottom layers. I think it was artifacting on 3 layers.

**Comment by andrewboktor** [comment]
> Hi, I resolved this by accident. I increased the count of bottom layers and the artifacting I had went away after re-slicing. This has worked for me both times I have had this problem. The second time I used 6 bottom layers. I think it was artifacting on 3 layers.

Is this comment relevant to this issue or was that posted by mistake? Can you elaborate it is indeed in the right place?

**Comment by Tomsko567** [comment]
> > Hi, I resolved this by accident. I increased the count of bottom layers and the artifacting I had went away after re-slicing. This has worked for me both times I have had this problem. The second time I used 6 bottom layers. I think it was artifacting on 3 layers.
> 
> Is this comment relevant to this issue or was that posted by mistake? Can you elaborate it is indeed in the right place?

Hi, this is the first time I have used a forum so apologies if I am not doing this right.

Yes, I found this page when I was having the same problem slicing in spiral mode when I was getting a  strange artifact/seam as described. When I used 3 base layers the artifact/seam was there. When I increased to 6 base layers the issue simply disappeared. Screen shots attached…

![Screenshot 2024-03-20 095145](https://github.com/bambulab/BambuStudio/assets/163763187/d32f47b7-079d-4905-9fcc-486aa2fd4978)
![Screenshot 2024-03-20 095231](https://github.com/bambulab/BambuStudio/assets/163763187/c95873e0-4b9e-4508-aa63-e5574a012c7f)
